### PR TITLE
Make exception messages more generic

### PR DESCRIPTION
**Pull request by rdlowrey**
Since the exceptions don't have to be specifically tied to promise resolution I believe it makes more sense to make the default messages more generic.

### CODE REVIEW

File: lib/Observer.php [hunk 1]
```diff
@@ -122,7 +122,7 @@ public function next(): Promise {
 
         ++$this->position;
 
-        if (\array_key_exists($this->position, $this->values)) {
+        if (isset($this->values[$this->position]) || \array_key_exists($this->position, $this->values)) {
```
Comment (bwoebi) [inline lib/Observer.php]: uh, wrong PR?

File: lib/TimeoutException.php [hunk 2]
```diff
@@ -7,6 +7,6 @@ class TimeoutException extends \Exception {
      * @param string|null $message
      */
     public function __construct(string $message = null) {
-        parent::__construct($message ?: "Promise resolution timed out");
+        parent::__construct($message ?: "Operation timed out");
```
Comment (bwoebi) [inline lib/TimeoutException.php]: Uhm, Amp\TimeoutException is used exclusively for promise timeouts…!?

File: lib/Observer.php (hunk 1 above)
Comment (rdlowrey) [inline lib/Observer.php]: Yeah this was a derp. Ignore.

File: lib/TimeoutException.php (hunk 2 above)
Comment (rdlowrey) [inline lib/TimeoutException.php]: Just because it doesn't have to be. It could be used for anything anyone wants it to be used for ...

Comment (rdlowrey) [inline lib/TimeoutException.php]: Otherwise it should be called `PromiseTimeoutException` IMO

Comment (bwoebi) [inline lib/TimeoutException.php]: In general you shall use library specific exceptions anyway, so no reason to.

Comment (bwoebi) [inline lib/TimeoutException.php]: Dunno, Amp is _the_ Promise library. no need to be more specific here. If you catch, you shall feel free to `use Amp\TimeoutException as PromiseTimeoutException;`

Comment (kelunik) [inline lib/TimeoutException.php]: `Operation timed out` is fine IMO if not even better.

Comment (staabm) [inline lib/TimeoutException.php]: Agree

File: lib/MultiReasonException.php [hunk 3]
```diff
@@ -11,7 +11,7 @@ class MultiReasonException extends \Exception {
      * @param string|null $message
      */
     public function __construct(array $reasons, string $message = null) {
-        parent::__construct($message ?: "Too many promises were rejected");
+        parent::__construct($message ?: "Multiple errors encountered");
```
Comment (kelunik) [inline lib/MultiReasonException.php]: I think we should stringify at least the exception names and messages into the message, otherwise they get lost.

**Pull request merged**